### DISCUSSION TITLE
Can a Data Package reference other Data Packages as resources ?

### DISCUSSION THREAD

**Discussion by jeanpommier** [frictionlessdata/frictionlessdata.io#621] [Q&A]
Hi,
I'm considering adopting Data Packages for a quite complex data structure, where I have nested structure and the need for metadata at different levels. 
I'm a Data Package newbie, sorry if I'm asking silly questions. I not see any use case nesting Data Packages (i.e a Data Package, that, in its resources definition, references other Data Packages, stored in subdirectories of this data package). Is it considered / possible ? How would such nested packages be referred in the resources list ?
Thanks

**Comment by rufuspollock** [comment]
@jeanpommier that's definitely possible to do. It does sound a bit like the root data package would be more like a catalog - see https://github.com/frictionlessdata/specs/issues/37 (we were going to write this up as a pattern but haven't done so).

The other alternative is just to make everything resources below the root and add the necessary metadata there.

If you have more detail on the exact use case that would be useful 😄

**Reply by jeanpommier** [reply]
Thanks @rufuspollock 
Yes, you're right. I'm  working on a file-based implementation for https://docs.riskdatalibrary.org/overview.html
The schema is quite complex, but I think we can manage to flatten it to a hierarchical file structure. Most data will be geospatial files, with accompanying metadata, but we also have some tabular data. Metadata information will happen at the various hierarchical levels. The whole set of files will constitute a Risk dataset.

 I would not see it as catalog, since the next step will indeed be to create a catalog of such datasets.
I'm still uncertain as wether the Data Package concept might apply or not.

Do you have examples of such nested data packages ?

**Reply by rufuspollock** [reply]
Just to note Jean and I had a call with others from the Risk team. Thought was that the best approach was to keep it simple to start with and limit the nesting for now.
